helm & metrics: fix hive config properties of type stringslice

mhofstetter opened the pull request:
Hive config flags of type `StringSlice` are no longer split by space. Instead, they are split by comma. This leads to issues for config values that have been lately migrated to a Hive cell and didn't change this in the Helm files.

Therefore, this commit changes the affected config values to pass them as a comma-separated string.

* `prometheus.metrics` -> `metrics`
* `prometheus.controllerGroupMetrics` -> `controller-group-metrics`

Fixes: https://github.com/cilium/cilium/pull/26850, https://github.com/cilium/cilium/issues/28771

Thanks @giorio94 for catching this!

mhofstetter committed:
helm & metrics: fix hive config properties of type stringslice

Hive config flags of type `StringSlice` are no longer split by space.
Instead, they are split by comma. This leads to issues for config values
that have been lately migrated to a Hive cell and didn't change this in the
Helm files.

Therefore, this commit changes the affected config values to pass them as
a comma-separated string.

* `prometheus.metrics` -> `metrics`
* `prometheus.controllerGroupMetrics` -> `controller-group-metrics`

Fixes: #26850, #28771

Suggested-by: Marco Iorio <marco.iorio@isovalent.com>

Signed-off-by: Marco Hofstetter <marco.hofstetter@isovalent.com>

giorio94 reviewed (commented):
Thanks @mhofstetter for addressing this! Overall the fix looks good to me, just two minor things:
* IIRC there's also a mention of "space separated value" in the values file about the metrics, which could also use an update.
* The metrics fix should be split in a separate PR, as it needs to be backported to v1.14

sayboras reviewed (approved):
LGTM for helm changes :heavy_check_mark:

learnitall reviewed (approved):
LGTM for docs!

giorio94 reviewed (approved):
Thanks! Just a note that controller group metrics weren't present in v1.14 AFAIR, which might cause conflicts if we backport this (although they would be easy to resolve).

asauber commented on install/kubernetes/cilium/values.yaml.tmpl:
```diff
@@ -1926,8 +1926,7 @@ prometheus:
   # -- Metrics that should be enabled or disabled from the default metric list.
-  # The list is expected to be separated by a space. (+metric_foo to enable
-  # metric_foo , -metric_bar to disable metric_bar).
+  # (+metric_foo to enable metric_foo , -metric_bar to disable metric_bar).
```
It should be noted here that the value is an array, with the example matching that format.

```
    metrics:
      - +metric_foo
      - -metric_bar
  
This enables `metric_foo` and disables `metric_bar`.
```

mhofstetter replied:
Updated. Changed the type of the property `metrics` from `string` to `array` by setting the default from `~` to `[]`.

This shouldn't change any behaviour. It's mainly that it's displayed properly in the docs too.

asauber reviewed (changes requested):
This documentation update should be made before merging

https://github.com/cilium/cilium/pull/28919#discussion_r1383628055

mhofstetter committed:
helm: change type of metrics from string to array

This commit changes the default type of the Helm property
`metrics` from string to array. This way it's properly
displayed in the docs.

Signed-off-by: Marco Hofstetter <marco.hofstetter@isovalent.com>

joestringer reviewed (changes requested):
We should solve this comprehensively ref. https://github.com/cilium/cilium/issues/29210 .
